dali/operators/image/color/color_space_conversion.h

```diff
@@ -26,23 +26,47 @@ template <typename Backend>
 class ColorSpaceConversion : public Operator<Backend> {
  public:
   inline explicit ColorSpaceConversion(const OpSpec &spec)
-    : Operator<Backend>(spec)
-    , input_type_(spec.GetArgument<DALIImageType>("image_type"))
-    , output_type_(spec.GetArgument<DALIImageType>("output_type")) {
+      : Operator<Backend>(spec),
+        input_type_(spec.GetArgument<DALIImageType>("image_type")),
+        output_type_(spec.GetArgument<DALIImageType>("output_type")),
+        in_nchannels_(NumberOfChannels(input_type_)),
+        out_nchannels_(NumberOfChannels(output_type_)) {
   }
 
  protected:
-  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override {
-    return false;
+  bool CanInferOutputs() const override { return true; }
+  bool SetupImpl(std::vector<OutputDesc> &output_desc,
+                 const workspace_t<Backend> &ws) override {
+    output_desc.resize(1);
+    const auto &input = ws.template InputRef<Backend>(0);
+    auto in_sh = input.shape();
+    auto ndim = in_sh.sample_dim();
+    int nsamples = in_sh.num_samples();
+    auto in_layout = input.GetLayout();
+    int channel_dim = ndim - 1;
```

Comment:
I think that the way it's written now doesn't convey the idea very well.
The actual channel dimension is what we find in the layout (if any) and then we should check that it meets the constraints. If we allow planar layouts in the future, we'll simply drop the enforce.
```suggestion
    int channel_dim = in_layout.contains('C') ? in_layout.find('C') : ndim - 1;
    DALI_ENFORCE(channel_dim == ndim - 1, make_string("Channel dimension should be the last in the layout. Got ", in_layout));
```

Reply:
Will do (the second suggestion)